File: movingChips/movingchips.py
T = int(input().strip())
for _ in range(T):
    n = int(input().strip())
    a = list(map(int, input().strip().split())) # the list of chips
    first_chip = a.index(1)
    last_chip = n - 1 - a[::-1].index(1)
    print(a[first_chip:last_chip+1].count(0))

File: movingChips/movingchips.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int t;
    std::cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        vector<int> a(n);
        for (int i = 0; i < n; ++i)
        {
            cin >> a[i];
        }
        auto first_chip = std::find(a.begin(), a.end(), 1);
        auto last_chip = std::find(a.rbegin(), a.rend(), 1);
        int free_cells = std::count(first_chip, a.end() - (last_chip - a.rbegin()), 0);
        cout << free_cells << std::endl;
    }
    return 0;
}